enhancements/cvo/object-removal-manifest-annotation.md

```diff
@@ -0,0 +1,137 @@
+
+## Summary
+
+Currently CVO managed object removal is handled by jobs.
+Jobs have many immutable components, so they aren't a great match for CVO managed manifests.
+This enhancement replaces that approach with a new manifest annotation requesting the CVO delete the in-cluster object instead of creating/reconciling it.
+This will provide a more straightforward way to remove content.
+
+## Motivation
+
+There should be a straightforward way to remove CVO managed objects.
+
+### Goals
+
+* Users will be able to remove any of the currently managed CVO objects by simply modifying the object's existing manifest and adding the new delete annotation.
+
+### Non-Goals
+
+* This enhancement does not add any new objects for managment by CVO, therefore the new delete annotation only applies to the currently supported manifests and any that may be added in the future.
+
+## Proposal
+
+When the following annotation appears in a CVO supported manifest and is set to "true" the associated object will be removed from the cluster by the CVO. Values other than `true` will be ignored.
+
+```
+apiVersion: apps/v1
+...
+metadata:
+...
+  annotations:
+    release.openshift.io/delete: "true" 
+```
+
+### User Stories
+
+#### Story 1
+
+Remove the cluster-autoscaler-operator deployment.
+The existing cluster-autoscaler-operator deployment manifest 0000_50_cluster-autoscaler-operator_07_deployment.yaml is modified to contain the delete annotation:
+```
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cluster-autoscaler-operator
+  namespace: openshift-machine-api
+  annotations:
+    release.openshift.io/delete: "true"
+...
+```
+
+#### Story 2
+
+In release 4.5 two jobs, openshift-service-catalog-controller-manager-remover and openshift-service-catalog-apiserver-remover, were created to remove the Service Catalog.
+For release 4.6 these jobs and all their supporting cluster objects should be removed.
+This enhancement allows a user to accomplish this in a very simple manner by modifying all the manifests associated with these two jobs to contain the delete annotation.
+In this case the manifests to be modified are:
+* 0000_50_cluster-svcat-apiserver-operator_04_roles.yaml
+* 0000_50_cluster-svcat-apiserver-operator_05_serviceaccount.yaml
+* 0000_50_cluster-svcat-controller-manager-operator_06_roles.yaml
+* 0000_50_cluster-svcat-controller-manager-operator_07_serviceaccount.yaml
+* 0000_90_cluster-svcat-apiserver-operator_01_remover_job.yaml
+* 0000_90_cluster-svcat-controller-manager-operator_01_remover_job.yaml
+
+### Implementation Details/Notes/Constraints
+
+CVO will use its existing logic to discover and identify given manifests and to handle any associated errors.
+Once the manifest has been properly identified CVO will check for the delete annotation release.openshift.io/delete.
+If found and set to "true" the associated object will be removed from the cluster.
```

Comment:
Also if the object does not exist in the cluster CVO will not return error or log any error in the log file, right?

Reply:
Good point but do we at least want to log a warning "hey, someone is trying to delete something that doesn't exist"? Maybe they typo'ed the name, namespace, etc

Reply:
> Good point but do we at least want to log a warning "hey, someone is trying to delete something that doesn't exist"?

I don't think it's a warning, it's just our usual "reconciling manifest 5 of 400... doesn't exist, great, moving on".

Reply:
Nothing new added here since existing CVO logic will note condition.